hail/python/hail/utils/interval.py

```diff
@@ -26,52 +27,45 @@ class Interval(object):
     @typecheck_method(start=anytype,
                       end=anytype,
                       includes_start=bool,
-                      includes_end=bool)
-    def __init__(self, start, end, includes_start=True, includes_end=False):
-        from hail.expr.expressions import impute_type, unify_types_limited
-        start_type = impute_type(start)
-        end_type = impute_type(end)
-        point_type = unify_types_limited(start_type, end_type)
-
+                      includes_end=bool,
+                      point_type=nullable(hail_type))
+    def __init__(self, start, end, includes_start=True, includes_end=False, point_type=None):
         if point_type is None:
```

Comment:
Why did you change the signature of the constructor to take `point_type`? Are you trying to eliminate the checks if the user passes what the point_type is?

Reply:
This is to preserve the interface - we need to be able to construct intervals in python that might contain None, and type imputation will fail on None.

(I think)

Reply:
Yes, what Tim said.  I'm removing Interval._from_java, which took took the point type instead of inferring it.  The calls to _from_java turned into calls to the constructor, so I need to pass in the point type in those cases to preserve the old behavior.  I think adding an argument with a default isn't a breaking change.  I could have an alternate constructor in this case but I think this is clearer.